File: itsm_backend/src/main/java/itsm/itsm_backend/message/MessageResponse.java
package itsm.itsm_backend.message;

import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class MessageResponse {

    private Long id;
    private String content;
    private MessageType type;
    private MessageState state;
    private Integer senderId;
    private Integer receiverId;
    private LocalDateTime createdAt;
    private byte[] media;

}

File: itsm-ui/src/app/models/message-response.ts
export interface MessageResponse {
  content?: string;
  createdAt?: string;
  id?: number;
  media?: Array<string>;
  receiverId?: number;
  senderId?: number;
  state?: 'SENT' | 'SEEN';
  type?: 'TEXT' | 'IMAGE' | 'AUDIO' | 'VIDEO';
}